pkg/client/clientcmd/overrides.go

```diff
@@ -109,8 +111,12 @@ func RecommendedConfigOverrideFlags(prefix string) ConfigOverrideFlags {
 
 // BindAuthInfoFlags is a convenience method to bind the specified flags to their associated variables
 func BindAuthInfoFlags(authInfo *clientcmdapi.AuthInfo, flags *pflag.FlagSet, flagNames AuthOverrideFlags) {
-	// TODO short flag names are impossible to prefix, decide whether to keep them or not
-	flags.StringVarP(&authInfo.AuthPath, flagNames.AuthPath, "a", "", "Path to the auth info file. If missing, prompt the user. Only used if using https.")
+	// TODO short flag names are impossible to prefix.  code gets cleaner if we remove them
+	if len(flagNames.AuthPathShort) > 0 {
+		flags.StringVarP(&authInfo.AuthPath, flagNames.AuthPath, flagNames.AuthPathShort, "", "Path to the auth info file. If missing, prompt the user. Only used if using https.")
```

Comment:
Why the repetition in the `if` and `else` clauses? This is bound to get out of sync at some point...

As far as I can tell, if you call `flags.StringVarP` and pass it a "" as short option it will simply keep no short option for it, so why not just always call it with `flagNames.AuthPathShort` considering it's already "" in the case you don't want a short flag for it?